src/trousse/dataset.py

```diff
@@ -901,6 +912,83 @@ def to_file(self, filename: Union[Path, str], overwrite: bool = False) -> None:
             except KeyError as e:
                 logging.error(f"Exporting data unsuccessful: \n{e}")
 
+    def fillna(
+        self,
+        columns: Union[List[str], str],
+        value: Any,
+        derived_columns: Union[List[str], str] = None,
+        inplace: bool = False,
+    ) -> Optional["Dataset"]:
+        """Fill NaN values in ``columns`` columns with value ``value``.
+
+        By default NaNs are filled in the original columns. To store the result of filling
+        in other columns, ``derived_columns`` parameter has to be set with the name of
+        the corresponding column names.
+
+        Parameters
+        ----------
+        columns : Union[List[str], str]
+            Names of the columns with NaNs to be filled
+        value : Any
+            Value used to fill the NaNs
+        derived_columns : Union[List[str], str], optional
+            Names of the columns where to store the filling result. Default is None,
+            meaning that NaNs are filled in the original columns.
+        inplace : bool, optional
+            Whether to modify the current Dataset or return a new instance. Default False,
+            meanining that a new instance of Dataset will be returned.
+
+        Returns
+        -------
+        Optional[Dataset]
+            The new Dataset with NaNs filled if ``inplace=False`` or None otherwise.
+
+        Raises
+        ------
+        ValueError
+            If the number of columns to be filled is different from the number of the
+            columns where to store the result (if ``derived_columns`` is not None)
+        """
+        if not inplace:
+            dataset = self._dataset_copy
+        else:
+            dataset = self
+
+        if not isinstance(columns, list):
+            columns = [columns]
+
+        if derived_columns:
+            if not isinstance(derived_columns, list):
+                derived_columns = [derived_columns]
+
+            if len(columns) != len(derived_columns):
+                raise ValueError(
+                    f"The number of derived_columns ({len(derived_columns)}) "
+                    f"must be equal to the number of columns ({len(columns)})"
+                )
+
+            for column, derived_column in zip(columns, derived_columns):
+                filled_col = dataset.df[column].fillna(value, inplace=False)
+                dataset._df[derived_column] = filled_col
+
+        else:
+            for column in columns:
+                dataset.df[column].fillna(value, inplace=True)
+
+        if not inplace:
+            return dataset
+
+    @property
+    def _dataset_copy(self) -> "Dataset":
+        """Return a deep copy of the Dataset instance.
+
+        Returns
+        -------
+        Dataset
+            Deep copy of the Dataset instance.
+        """
+        return copy.deepcopy(self)
```

Comment:
yes it does make sense. I was a bit worried about memory consumption, but I didn't think of a solution right away. I am implementing your solution here

Reply:
Yes. Compared to the others it was 0-priority issue, but I had this idea and we want our baby to look great

Reply:
I'm opening an issue about it. I'm not going to fix this in this PR because to do this _well_ I want to refactor Dataset and remove the possibility to create a Dataset from either an existing pd.DataFrame or from a CSV file (but this will create a lot of changes that are not related to this PR)

Reply:
Perfect! See #68